asn1sccserviceprovider.cpp

```diff
@@ -58,8 +60,8 @@ Asn1SccServiceProvider::Asn1SccServiceProvider(Settings::ServiceConstPtr setting
 
 Asn1SccServiceProvider::~Asn1SccServiceProvider()
 {
-    if (m_asn1sccService->state() != QProcess::NotRunning)
-        finalize();
+    if (m_keepRunning)
+        stop();
```

Comment:
I understand the name of variable 'm_keepRunning' as indicating if we want the service to be running, so this seem to be contradiciting.